base/array.jl

```diff
@@ -109,6 +137,19 @@ function isassigned(a::Array, i::Int...)
     ccall(:jl_array_isassigned, Cint, (Any, UInt), a, ii) == 1
 end
 
+"""
+    Base.selectorbytes(A::Array{T, N}) -> Array{UInt8, N}
+
+For an Array with `isbits` Union elements, return the "selector bytes" that indicate the index
+of the type for each array element, i.e. the type of `A[1]` is `Base.uniontypes(eltype(A))[Base.selectorbytes(A)[1] + 1]`.
+**NOTE**: The actual array selector bytes are returned, meaning if individual elements are modified, the original array will reflect those changes.
+Setting selector bytes to invalid or out-of-bounds type indexes may corrupt the original array.
+"""
+function selectorbytes(a::Array{T, N}) where {T, N}
+    isbitsunion(T) || return UInt8[]
+    return unsafe_wrap(Array{UInt8, N}, convert(Ptr{UInt8}, pointer(a)) + length(a) * elsize(a), size(a))
```

Comment:
I see what you mean. It's not used anywhere in Base, but it's something I planned on using in the Feather.jl package at least; the feather file-format stores arrays in contiguous chunks, with separate chunks for their "null-fields", essentially a bitmask, where each bit is set indicating whether an element is null or not. I would just want some way to "set the bits" when deserializing a feather file.